_posts/2019-02-28-into-the-darkness-of-viewmodel-lists-composition.md

```diff
@@ -13,15 +13,17 @@ tags:
 
 We had [a first look at what problem ViewModel Composition is](https://milestone.topics.it/2019/02/06/what-is-services-viewmodel-composition-again.html) designed to solve, then we analyzed what it means to [compose a single item](https://milestone.topics.it/2019/02/20/viewmodel-composition-maze.html), using a Product as sample model. It's now time to start exploring what it means to compose a list of Product ViewModels.
 
-## Dark times
+## Why can't we just...?
 
 The trap of composing lists of ViewModels is that we cannot apply the same technique used for single items. If we have a list of items, where each item in the list is composed by data coming from multiple services, we cannot compose each item. Composing each item one by one means that our list will turn into a scary `SELECT N+1` type of thing, but worse since it'll be over `HTTP`.
 
 Composing a list is a matter of doing 3 steps:
 
-- step 1: get the keys of the items that should be included in the list
-- step 2: get all the data for all the items in the list
-- step 3: compose
+- Step 1: get the keys of the items that should be included in the list
+- Step 2: get all the data for all the items in the list
+- Step 3: compose
+
+This reduces the number of HTTP calls to two and also reduces the load on our datastore since we can retrieve the details we need in one request (for each service).
```

Comment:
This might not be 100% accurate so correct it if necessary. My goal was to hammer home _why_ we're doing these three steps. You've identified a problem and proposed a solution but didn't explicitly say how it solves the problem.

Reply:
Yup, correct. Needs a small change, but all in all is correct.